src/ir/auth_logic/ast.h

```diff
@@ -112,61 +113,65 @@ class Fact;
 
 // CanSay corresponds to an expression of the form <principal> canSay Fact
 class CanSay {
-  public:
-    explicit CanSay(Principal principal, std::unique_ptr<Fact>& fact)
+ public:
+  explicit CanSay(Principal principal, std::unique_ptr<Fact>& fact)
       : principal_(principal), fact_(std::move(fact)) {}
-    const Principal& principal() const { return principal_; }
-    const Fact* fact() const { return fact_.get(); }
-  private:
-    Principal principal_;
-    std::unique_ptr<Fact> fact_;
+  const Principal& principal() const { return principal_; }
+  const Fact* fact() const { return fact_.get(); }
+
+ private:
+  Principal principal_;
+  std::unique_ptr<Fact> fact_;
 };
 
-// Fact corresponds to either a base fact or a an expression of the form 
-// <principal> canSay 
+// Fact corresponds to either a base fact or a an expression of the form
+// <principal> canSay <Fact>
+using FactVariantType = std::variant<BaseFact, std::unique_ptr<CanSay>>;
 class Fact {
  public:
   explicit Fact(BaseFact base_fact) : value_(std::move(base_fact)) {}
-  explicit Fact(std::unique_ptr<CanSay> can_say)
-    : value_(std::move(can_say)) {}
+  explicit Fact(std::unique_ptr<CanSay> can_say) : value_(std::move(can_say)) {}
+  const FactVariantType& GetValue() const { return value_; }
 
  private:
-  std::variant<BaseFact, std::unique_ptr<CanSay>> value_;
+  FactVariantType value_;
 };
 
-// ConditionalAssertion the particular form of assertion that can have 
+// ConditionalAssertion the particular form of assertion that can have
 // conditions which is:
 // <fact> :- <flat_fact_1> ... <flat_fact_n>
 class ConditionalAssertion {
-  public:
-    explicit ConditionalAssertion(Fact lhs, std::vector<BaseFact> rhs)
+ public:
+  explicit ConditionalAssertion(Fact lhs, std::vector<BaseFact> rhs)
       : lhs_(std::move(lhs)), rhs_(std::move(rhs)) {}
-    const Fact& lhs() const { return lhs_; }
-    const std::vector<BaseFact>& rhs() const { return rhs_; }
-  private:
-    Fact lhs_;
-    std::vector<BaseFact> rhs_;
+  const Fact& lhs() const { return lhs_; }
+  const std::vector<BaseFact>& rhs() const { return rhs_; }
+
+ private:
+  Fact lhs_;
+  std::vector<BaseFact> rhs_;
 };
 
 // Assertions can have two forms:
 //  - Facts
 //  - Conditional assertions
+using AssertionVariantType = std::variant<Fact, ConditionalAssertion>;
```

Comment:
Similar to the other public `using` alias.

Reply:
See other thread.